spring-session/src/main/java/org/springframework/session/AbstractIndexResolver.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2014-2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.session;
+
+/**
+ * Abstract base class for {@link IndexResolver} implementations.
+ *
+ * @param <S> the type of Session being handled
+ * @author Vedran Pavic
+ * @since 1.3.0
+ */
+public abstract class AbstractIndexResolver<S extends Session> implements IndexResolver<S> {
+
+	public String resolve(S session, String indexName) {
+		if (!supportedIndexName().equals(indexName)) {
+			return null;
+		}
+		return doResolve(session);
+	}
+
+	/**
+	 * Indicate supported index name.
+	 * @return the index name
+	 */
+	protected abstract String supportedIndexName();
```

Comment:
Well, this is basically just an abstract base class that simplifies implementing single-purpose ```IndexResolver```. Nothing in the ```IndexResolver``` contract itself prevents the implementations from supporting multiple indexes, as demonstrated by ```CompositeIndexResolver```.